src/Swarrot/Processor/XDeath/XDeathMaxCountProcessor.php

```diff
@@ -0,0 +1,180 @@
+                'x_death_max_count_log_levels_map' => array(),
+                'x_death_max_count_fail_log_levels_map' => array(),
+            ))
+            ->setAllowedTypes('x_death_max_count', 'int')
+            ->setAllowedTypes('x_death_max_count_log_levels_map', 'array')
+            ->setAllowedTypes('x_death_max_count_fail_log_levels_map', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, array $options)
+    {
+        try {
+            return $this->processor->process($message, $options);
+        } catch (\Throwable $e) {
+            return $this->handleException($e, $message, $options);
+        } catch (\Exception $e) {
+            return $this->handleException($e, $message, $options);
+        }
+    }
+
+    /**
+     * @param $exception
+     * @param Message $message
+     * @param array $options
+     *
+     * @return mixed
+     */
+    private function handleException($exception, Message $message, array $options)
+    {
+        $headers = $message->getProperties();
+        if (isset($headers['headers']['x-death'], $options['x_death_max_count'])) {
+            $xDeathHeaders = $headers['headers']['x-death'];
+            // PhpAmqpLib compatibility
+            if (class_exists('PhpAmqpLib\Wire\AMQPArray') && $headers['headers']['x-death'] instanceof AMQPArray) {
+                $xDeathHeaders = $headers['headers']['x-death']->getNativeData();
+            }
+            $queueName = $this->messageProvider->getQueueName();
+            $queueXDeathHeader = null;
+            foreach ($xDeathHeaders as $xDeathHeader) {
+                if (isset($xDeathHeader['queue']) && $xDeathHeader['queue'] === $queueName) {
+                    $queueXDeathHeader = $xDeathHeader;
+                    break;
+                }
+            }
+
+            if (null === $queueXDeathHeader) {
+                $this->logException(
+                    $exception,
+                    sprintf(
+                        '[XDeathMaxCount] No x-death header found for queue name "%s". Do nothing.',
+                        $queueName
+                    ),
+                    $options['x_death_max_count_fail_log_levels_map']
+                );
+            }
+
+            if (isset($queueXDeathHeader['count'])) {
+                if ($queueXDeathHeader['count'] >= $options['x_death_max_count']) {
+                    $this->logException(
+                        $exception,
+                        sprintf(
+                            '[XDeathMaxCount] Max count reached. %d/%d attempts. Execute the configured callback.',
+                            $queueXDeathHeader['count'],
+                            $options['x_death_max_count']
+                        ),
+                        $options['x_death_max_count_fail_log_levels_map']
+                    );
+
+                    if (null !== $return = call_user_func($this->callback, $exception, $message, $options)) {
```

Comment:
please use a fully-qualified call here (i.e. `\call_user_func`), which will allow recent PHP versions to optimize this call by avoiding to create a method call for `call_user_func` itself (replacing it by a function call at compile time)

Reply:
:+1: